pkg/virt-controller/ipamclaims/ipamclaims_test.go

```diff
@@ -0,0 +1,337 @@
+			libvmi.WithNetwork(libvmi.MultusNetwork("absent", "absent-net")),
+			libvmi.WithInterface(virtv1.Interface{Name: defaultNamespace}),
+			libvmi.WithInterface(virtv1.Interface{Name: redNetworkLogicalName}),
+			libvmi.WithInterface(virtv1.Interface{Name: blueNetworkLogicalName}),
+			libvmi.WithInterface(virtv1.Interface{Name: "absent", State: virtv1.InterfaceStateAbsent}),
+		)
+		vmi.UID = vmiUID
+	})
+
+	Context("With allowPersistentIPs enabled in the NADs", func() {
+		var ipamClaimsManager *ipamclaims.IPAMClaimsManager
+
+		BeforeEach(func() {
+			persistentIPs := map[string]struct{}{redNetworkNadName: {}, blueNetworkNadName: {}}
+			Expect(createNADs(networkClient, vmi.Namespace, vmi.Spec.Networks, persistentIPs)).To(Succeed())
+
+			ipamClaimsManager = ipamclaims.NewIPAMClaimsManager(networkClient, ipamClaimsClient)
+		})
+
+		It("should create the expected IPAMClaims", func() {
+			ownerRef := &v1.OwnerReference{
+				APIVersion:         "kubevirt.io/v1",
+				Kind:               "VirtualMachine",
+				Name:               vmi.Name,
+				UID:                vmUID,
+				Controller:         pointer.P(true),
+				BlockOwnerDeletion: pointer.P(true),
+			}
+			Expect(ipamClaimsManager.CreateNewPodIPAMClaims(vmi, ownerRef)).To(Succeed())
+
+			ipamClaimsList, err := ipamClaimsClient.K8sV1alpha1().IPAMClaims(vmi.Namespace).List(
+				context.Background(),
+				v1.ListOptions{},
+			)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ipamClaimsList.Items).To(HaveLen(2))
+			assertIPAMClaim(ipamClaimsList.Items[0], vmi.Namespace, vmi.Name, blueNetworkLogicalName, "pod16477688c0e", "VirtualMachine")
+			assertIPAMClaim(ipamClaimsList.Items[1], vmi.Namespace, vmi.Name, redNetworkLogicalName, "podb1f51a511f1", "VirtualMachine")
+		})
+
+		Context("When IPAMClaims already exist", func() {
+			var ownerRef *v1.OwnerReference
+
+			BeforeEach(func() {
+				ownerRef = &v1.OwnerReference{
+					APIVersion:         "kubevirt.io/v1",
+					Kind:               "VirtualMachine",
+					Name:               vmi.Name,
+					UID:                vmUID,
+					Controller:         pointer.P(true),
+					BlockOwnerDeletion: pointer.P(true),
+				}
+				Expect(ipamClaimsManager.CreateNewPodIPAMClaims(vmi, ownerRef)).To(Succeed())
+
+				ipamClaimsList, err := ipamClaimsClient.K8sV1alpha1().IPAMClaims(vmi.Namespace).List(
+					context.Background(),
+					v1.ListOptions{},
+				)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(ipamClaimsList.Items).To(HaveLen(2))
+			})
+
+			It("with the expected owner UID, should not fail re-creation/validation attempt", func() {
+				Expect(ipamClaimsManager.CreateNewPodIPAMClaims(vmi, ownerRef)).To(Succeed())
+
+				ipamClaimsList, err := ipamClaimsClient.K8sV1alpha1().IPAMClaims(vmi.Namespace).List(
+					context.Background(),
+					v1.ListOptions{},
+				)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(ipamClaimsList.Items).To(HaveLen(2))
```

Comment:
maybe we should compare them ? i.e. store the `ipamClaimsList` from the test setup, and ensure here they still exist with the same values.

Reply:
thought about it, sounds bit overkill imo, we know that we don't alter, either create or assert
i think we don't need to be that strict unless you insist

Reply:
I won't insist, but I think it is preferable. 

IIUC you'd just have to have a variable reachable both in the test and its setup; call it `expectedClaims` or `expectedIPAMClaims`. 

Then you can `Expect(ipamClaimsList.Items).To(ConsistOf(expectedClaims))` in the test. 

While I understand the "real" part of the test here is it throwing an error (or not), IMHO this become more readable. 

I won't insist though.